keps/973-workload-priority/README.md

```diff
@@ -0,0 +1,331 @@
+  - [Future works](#future-works)
+  - [Test Plan](#test-plan)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+In this proposal, a `WorkloadPriorityClass` is created.
+The `Workload` is able to utilize `WorkloadPriorityClass`.
+`WorkloadPriorityClass` is independent from pod's priority.
+The priority value is a part of the workload spec and is mutable.  
+In this document, the term "workload Priority" is used to refer
+to the priority utilized by Kueue controller for managing the queueing
+and preemption of workloads.  
+The term "pod Priority" is used to denote the priority utilized by the
+kube-controller for preempting pods and jobs.
+
+## Motivation
+
+We want to make the priority of the workload customizable according to user's needs.
+By defining `WorkloadPriorityClass`, we aim to enable various cutomizations only for Kueue's `Workload`.
+
+### Goals
+
+Implement `WorkloadPriorityClass`.
+`Workload` can utilize `WorkloadPriorityClass`.
+CRDs like Job, MPIJob etc specify the `WorkloadPriorityClass` through labels.
+
+### Non-Goals
+
+Using existing k8s `PriorityClass` for Workload's Priority is not recommended.
+`WorkloadPriorityClass` doesn't implement all the features of the k8s `PriorityClass`
+because some fields on the k8s `PriorityClass` are not relevant to Kueue.
+
+## Proposal
+
+In this proposal, `WorkloadPriorityClass` is defined.
+The `Workload` is able to utilize this `WorkloadPriorityClass`.
+`WorkloadPriorityClass` is independent from pod's priority.
+The priority value will be part of the workload spec and be mutable.
+CRDs like Job, MPIJob etc specify the `WorkloadPriorityClass` through labels.
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories
+
+Kueue issue [973](https://github.com/kubernetes-sigs/kueue/issues/973) provides details on the initial feature implementation.
+
+#### Story 1
+
+In an organization, admins want to set a lower priority for development workloads and a higher priority for production workloads.
+In such cases, they create two `WorkloadPriorityClass` and apply each one to the respective workloads.
+
+#### Story 2
+
+An organization desires to modify the priority of workloads that remain inactive for a specific duration.
+By developing a custom controller to manage Priority value of `WorkloadPriorityClass`, this expectation can be met.
```

Comment:
They shouldn't change the value in the WorkloadPriorityClass, as that would affect all Workloads. But they can change the value in the Workload spec.